### PR TITLE
Added r-qgisprocess as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/staged-recipes @conda-forge/help-r ready for review!

File: recipes/r-qgisprocess/meta.yaml
Comment (mfansler) [inline recipes/r-qgisprocess/meta.yaml:60]: Split URLs

```suggestion
  home: https://r-spatial.github.io/qgisprocess/
  dev_url: https://github.com/r-spatial/qgisprocess
```

File: recipes/r-qgisprocess/meta.yaml
Comment (mfansler) [inline recipes/r-qgisprocess/meta.yaml:61]: DESCRIPTION shows `GPL (>= 3)`

```suggestion
  license: GPL-3.0-or-later
```

File: recipes/r-qgisprocess/meta.yaml
Comment (mfansler) [inline recipes/r-qgisprocess/meta.yaml:96]: The recipe appears not to address this. This should be added as `run:` requirement.

One can also see it failing in the CI test (though it exits with 0):

<details>

```r
R version 4.2.3 (2023-03-15) -- "Shortstop Beagle"
Copyright (C) 2023 The R Foundation for Statistical Computing
Platform: x86_64-conda-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library('qgisprocess')
Attempting to load the package cache ... No cache found.
Will try to reconfigure qgisprocess and build new cache ...

>>> PROBLEM encountered: couldn't build package cache! <<<
The 'qgis_process' command-line utility was either not found or
did not fulfil the needs to build the package cache.
Please run `qgis_configure()` to fix this and rebuild the cache.
See its documentation if you need to preset the path of qgis_process.
If the problem persists, make sure that you correctly installed QGIS
for your operating system using the instructions at
https://download.qgis.org./
> 
>
```

</details>

I would recommend that some integration test be run - something that will actually fail if the QGIS is not usable. Does this package has `testthat` tests that could be run?

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for adding this! 

Since this package is an interface to QGIS, it should depend on that and ensure that it interfaces properly with the version we deliver. At minimum, some function in the package that calls out to QGIS should be tested. If there are `testthat` tests that we could run that would be ideal.

File: recipes/r-qgisprocess/meta.yaml
Comment (thewchan) [reply recipes/r-qgisprocess/meta.yaml:60]: Thanks, done!

File: recipes/r-qgisprocess/meta.yaml
Comment (thewchan) [reply recipes/r-qgisprocess/meta.yaml:61]: Thanks, done!

File: recipes/r-qgisprocess/meta.yaml
Comment (thewchan) [reply recipes/r-qgisprocess/meta.yaml:96]: I'll add qgis first; I'll look to see if they have tests in place

**Comment by mfansler** [comment]
Conda Forge build script doesn't install tests by default, so `test_package` won't do the trick. You could install the tests to the build, but I that adds space to the final build (I used to do this, but have tried to move away from it). Instead, here are a couple examples for how to run the testhat tests without installing tests:

- [`r-vetiver`](https://github.com/conda-forge/r-vetiver-feedstock/blob/main/recipe/meta.yaml) - shows how to run the `tests/testthat.R` file, if that is configured to execute all the tests
- [`r-openssl`](https://github.com/conda-forge/r-openssl-feedstock/blob/main/recipe/meta.yaml) - shows how to run all the `tests/testthat/*` tests when the `testthat.R` file does not orchestrate all tests

**Comment by thewchan** [comment]
@mfansler When I install `qgis` via conda and `r-qgisprocess` via R/CRAN locally it was able to pick up `quisprocess` from path without problems, but here in the testing CI it's not able to for some reason.

**Comment by thewchan** [comment]
@mfansler okay so definitely it's not in path, `which` can't find it. This is strange, let me try including `qgis` in test dep.

**Comment by thewchan** [comment]
@mfansler no dice, not sure what else to try?

**Comment by mfansler** [comment]
Running on anvil image (`condaforge/linux-anvil-cos7-x86_64:latest`) locally, I definitely see `qgis_process` on PATH in activated environment. Rather, I see:

```bash
$ qgis_process --help
qgis_process: error while loading shared libraries: libGL.so.1: cannot open shared object file: No such file or directory
```

which implies it's not a PATH issue, but we need to [add libGL support to the runner](https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl).

**Comment by thewchan** [comment]
@mfansler thanks for checking this, hoping this will shed some light

**Comment by thewchan** [comment]
It seems like maybe this should also be fixed in the `qgis` feedstock?

**Comment by mfansler** [comment]
> It seems like maybe this should also be fixed in the `qgis` feedstock?

First, just making sure our runner is at parity with theirs. But yeah, it might be useful to add a test of `qgis_process --help` to [their test script](https://github.com/conda-forge/qgis-feedstock/blob/bcc489761fdb6d7ec865f745a990d36496028f9f/recipe/run_test.sh#L10).

**Comment by mfansler** [comment]
Well, **linux-64** is passing. The **osx-64** is still not being detected, however, the artifact is built, so maybe that can be tested locally if it works for an end-user. The **win-64** is recognized, but is having some parsing error - might be good enough for now, since most tests pass.

Technically, this whole thing is going to build as `noarch` (**linux-64** build), so it might be satisfactory to see here that **linux-64** is working.

**Comment by mfansler** [comment]
Turns out **osx-*** builds of `qgis` don't properly expose `qgis_process`. I tried locally with the `noarch` artifact here and I needed to add a symlink to get it working. I suggest we wait until that fix goes through, then we can test this again before merging.

**Review by mfansler — APPROVED** [review]

**Pull request merged**
